src/self_update.sh

```diff
@@ -0,0 +1,209 @@
+include "${KW_LIB_DIR}/kwlib.sh"
+include "${KW_LIB_DIR}/kwio.sh"
+include "${KW_LIB_DIR}/help.sh"
+
+declare -gA options_values
+
+function self_update_main()
+{
+  local target_branch='master'
+  local path_to_tmp_dir
+  local cmd
+  local flag
+  local ret
+
+  parse_self_update_options "$@"
+  if [[ "$?" != 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    self_update_help
+    return 22 # EINVAL
+  fi
+
+  if [[ -n "${options_values['UNSTABLE']}" ]]; then
+    target_branch='unstable'
+  fi
+
+  printf '%s\n' "Updating kw based on the branch ${target_branch}"
+  if [[ $(ask_yN 'Do you want to continue?') =~ '0' ]]; then
+    complain 'Update aborted.'
+    return 125 # ECANCELED
+  fi
+
+  flag="${options_values['TEST']}"
+  path_to_tmp_dir=$(mktemp --directory)
+  cmd="update_from_official_repo ${target_branch} ${path_to_tmp_dir}"
+  cmd_manager "$flag" "$cmd" > /dev/null
+
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain "Update from branch ${target_branch} failed"
+    return "$ret"
+  fi
+
+  success "Update based on the branch ${target_branch} successful!"
+}
+
+# Updates kw using the official kw repository.
+#
+# @target_branch Branch used to update
+# @path_to_tmp_dir Path to temp dir used for cloning repo
+# @test Indicates if it is a test or not
+#
+# Return:
+# In case of success return 0, otherwise, return the error
+# code.
+function update_from_official_repo()
+{
+  local target_branch="$1"
+  local path_to_tmp_dir="$2"
+  local flag
+  local original_dir="$PWD"
+  local cmd
+  local ret
+
+  cmd="cd ${path_to_tmp_dir}"
+  cmd_manager "$flag" "$cmd" > /dev/null
+  if [[ "$?" != 0 ]]; then
+    complain "Could not change to ${path_to_tmp_dir} dir"
+    return 2 # ENOENT
+  fi
+
+  cmd="git clone --quiet 'https://github.com/kworkflow/kworkflow.git'"
+  cmd_manager "$flag" "$cmd" > /dev/null
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain 'Could not clone kworkflow repository'
+    return "$ret"
+  fi
+
+  cmd="cd kworkflow"
+  cmd_manager "$flag" "$cmd" > /dev/null
+  if [[ "$?" != 0 ]]; then
+    handle_error 'Could not change to kworkflow repository' \
+      "${original_dir}" 1 'remove_kw_dir'
+    return "$?"
+  fi
+
+  cmd="git checkout --quiet ${target_branch}"
```

Comment:
btw, instead of `cd` you can use the `-C` option from git. This argument will simplify your code since you will not need to deal with `cd` in and out.